### DISCUSSION TITLE
Cruiser Wiki

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#490] [Show and tell]
We can discuss here anything about Cruiser Wiki:

- https://github.com/devemux86/cruiser/wiki

To format the text, see the [Basic writing and formatting syntax](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax).

**Comment by DIrgi** [comment]
Oh, I see and checked out the wiki access is already open for everyone. I'm already testing there a little bit.
I can see that on the right side, pages can be added - my proposal ist to use this as a kind of a table of contents
There we should define main topics and beyond the main topics there could be the several sub toipics

Main topics can be:

- Cruiser App setup
- How to Navigate
- or simply orienting on the given menus of cruiser having them as the first head menus in consideration
- 
<img src="https://github.com/devemux86/cruiser/assets/57968526/a92639b7-16ce-4d77-adcb-081d889b2b67" width="50%"/>

**Reply by bios4** [reply]
This is a screenshot from an old Cruiser version @DIrgi - I think we should not invest time to describe outdated UI already at the very beginning... 😉

**Reply by DIrgi** [reply]
Don‘t blame on the picture. As I said, I‘m testing and btw this shot is done on my PC. No matter which Cruiser Version. I‘m running there a virtual Android OS where I can use the cruiser app for testing purposes 😉
More convenient then switching between GitHub on desktop to cruiser on tablet 😃

**Comment by DIrgi** [comment]
Oh, that's bad. The pages cannot have sub menus. 
this will limit building up a structure enormously 😒

**Reply by bios4** [reply]
There seem to be possibilities - see my additions here: https://github.com/devemux86/cruiser/wiki/Unfortunately-NO-submenu--structure-possible

@devemux86 - maybe you could habe a look and potentially install the mentioned module (if it is still working, the found link is from 2018)!??

**Comment by devemux86** [comment]
As an example of what Wiki can do, we can see how the [c:geo Wiki](https://github.com/cgeo/cgeo/wiki) is structured.

**Reply by DIrgi** [reply]
yes and there are submenus and a menu structure

**Comment by devemux86** [comment]
I edited the Wiki to add a menu with main categories and sub-menus:
- Guides
- Menu
- Settings

Of course this menu is just a thought. :slightly_smiling_face:

**Reply by bios4** [reply]
That looks great so far, thank you!
From there we could start building the Wiki step by step... 👍

> As an example of what Wiki can do, we can see how the [c:geo Wiki](https://github.com/cgeo/cgeo/wiki) is structured.

This looks nice, but it also there the "Pages" menu is ordered just alphabetically.
I assume this won't work for the Cruiser documentation...

So are you planning to give this npm-module a try?

**Reply by devemux86** [reply]
It is indeed an old module, probably for the Wiki of its time. Also all these years Wiki has new features too.
I see that we can have an ordered structure and sub-menus at the menu.

So better let's start with the provided tools and see step by step?

**Reply by devemux86** [reply]
I added a link to the markdown syntax writing guide above in the [1st post](https://github.com/devemux86/cruiser/discussions/490#discussion-6488455).

**Reply by devemux86** [reply]
I added pages to each menu for easy writing.

Again, this menu is not fixed! It's just a suggestion of how we can get started.
You are free to change it and add / remove pages making it the way you think is best for users.

**Reply by DIrgi** [reply]
The suggestions is already very well, I like that.

**Reply by DIrgi** [reply]
I add for testing a collapse menu

**Reply by DIrgi** [reply]
And some pics ;-)

**Reply by devemux86** [reply]
Please check permissions for images that are not from the app.

**Reply by DIrgi** [reply]
These are absolute fictive ai pics , freshly produced 😉

**Reply by bios4** [reply]
Nice design!

What AI service did you use to create these?

**Reply by DIrgi** [reply]
The free one, Designer from Bing, Mikkisoft . Really crazy. Check this out, I did for eastern 😁

![image](https://github.com/devemux86/cruiser/assets/57968526/9c7325e9-d7c0-48cf-9ec8-95b25ae9b9d8)

or here, recently done: Fat biker, using navi 😂👍

![image](https://github.com/devemux86/cruiser/assets/57968526/07726b5e-fd1b-4971-acca-7e8539f2966e)

![image](https://github.com/devemux86/cruiser/assets/57968526/bb8bc911-b236-4f45-a718-4eb120c708f3)

![image](https://github.com/devemux86/cruiser/assets/57968526/9daf83e8-dbc3-404b-a622-aa31f17b6c1d)

ok, I better stop. This has nothing to do here.

**Reply by devemux86** [reply]
@DIrgi 

Are these made by AI ?!

**Reply by DIrgi** [reply]
Yes, within seconds just by using catchwords..crazy world…

**Comment by DIrgi** [comment]
btw, I think collapse menus are better - what do you guys think? 
check it out on the wiki page

**Reply by devemux86** [reply]
If the menus are not too many, it is probably better to expand them for easy viewing.

**Reply by bios4** [reply]
I also like the expanded menus better:

1) You can see directly what is available
2) Less clicks

**Comment by bios4** [comment]
@devemux86

Are you deleting the Discussions thread for a specific Cruiser release as soon as a new version is released?
Or are they still available, but just not "visible" in the [Announcements](https://github.com/devemux86/cruiser/discussions/categories/announcements) category?

Reason for asking: I thought about adding a "version history" of Cruiser to the [What's new](https://github.com/devemux86/cruiser/wiki/What%27s-new%3F)-section in the Wiki, and wanted to link the release articles to these entries (see example of version 3.0.42).

But I found out that currently only one thread is still available in [Announcements](https://github.com/devemux86/cruiser/discussions/categories/announcements) - the one of 3.0.42.
Threads for older versions are gone!?

**Reply by devemux86** [reply]
> Or are they still available, but just not "visible" in the [Announcements](https://github.com/devemux86/cruiser/discussions/categories/announcements) category?

They are "closed" to not be visible. Should we do this?
But their links work.

They can be found when search with the "is:closed" filter, like [here](https://github.com/devemux86/cruiser/discussions?discussions_q=is%3Aclosed+).

**Reply by bios4** [reply]
If the links work we can keep it like it is.
What do YOU think about linking the release threads in the Wiki?

> They can be found when search with the "is:closed" filter, like [here](https://github.com/devemux86/cruiser/discussions?discussions_q=is%3Aclosed+).

Just tried it - did you really not create an entry for 3.0.38 (3.0.35, 3.0.34) or are they somehow "vanished" to Nirvana? 😀

**Reply by devemux86** [reply]
>  What do YOU think about linking the release threads in the Wiki?

Sure, we can do this.

> did you really not create an entry for 3.0.38 (3.0.35, 3.0.34)

There were some intermediate releases with just fixes that are mentioned inside the main releases.

(I am thinking of changing the versions from the next release for easier understanding)

**Comment by 0709wiwiwi** [comment]
Nice to add in the wiki FAQ's  ?
https://github.com/devemux86/cruiser/wiki/Frequently-asked-questions-%28FAQ%29

- How to get back the undo redo buttons after closing them by tapping the "X"?

_Long press on the map and find undo redo at the bottom of the menu dialog._

- How to add a shaping point not causing a recalculation ?

 _Long press route and set point (shaping or via). Press OK without moving the map._

**Reply by devemux86** [reply]
@0709wiwiwi 

Please feel free to start adding content to the Wiki if you think it is useful to users.
We can always edit and improve it later.

**Comment by 0709wiwiwi** [comment]


**Reply by bios4** [reply]
> https://github.com/devemux86/cruiser/wiki/How-to%3F

Nice first contributions to the Wiki, thank you!

Should we keep this "How to?" page in place, or move the content to the (existing, yet empty) "Frequently asked questions (FAQ)" section (like you already suggested earlier @0709wiwiwi)?

Keep both "How to?" **AND** "FAQs" doesn't make much sense from my POV!??

**Reply by bios4** [reply]
I doubt that this will work out, but what do I know...

**Comment by MotoUKRider** [comment]
I've started the [Profile menu](https://github.com/devemux86/cruiser/wiki/Profile-menu) and [Routing settings](https://github.com/devemux86/cruiser/wiki/Routing-settings) pages...

**Reply by devemux86** [reply]
@MotoUKRider many thanks! :+1:

**Reply by MotoUKRider** [reply]
...and the [Route planning guide](https://github.com/devemux86/cruiser/wiki/Route-planning-guide) :world_map: 

I plan to re-install Cruiser on my spare (S9) phone for the full *New User Experience* so that I can take some notes to help with the "Startup" aspects of the Wiki.

Something I did notice, it's more a GitHub thing - there's a link to this forum in Cruiser's "About" menu but the mobile-browser (Firefox) doesn't show all the menus/sections, so it's not immediately obviously there's a Wiki unless you tap the ... button.

Once the Wiki is more populated, perhaps a "Help/Wiki" link in the App, too?

**Reply by devemux86** [reply]
@MotoUKRider thanks!

> Once the Wiki is more populated, perhaps a "Help/Wiki" link in the App, too?

Indeed, we can do this. :+1:

**Comment by bios4** [comment]
General comment / question to @devemux86...

To create screenshots for the Wiki, it would make sense to obfuscate your real GPS position for data protection reasons.

I have read in an app description (e.g. https://play.google.com/store/apps/details?id=com.lexa.fakegps) that the "real" GPS position cannot always be automatically restored after using such apps, but that workarounds (even with a duration of some days) are sometimes necessary.

This should be avoided at any cost...

Is there any recommended app to achieve this, Emux?

**Reply by devemux86** [reply]
I use my own simulator for mock locations and navigation, much more advanced than anything on Google Play. :slightly_smiling_face: 

Have you tried Lockito? https://play.google.com/store/apps/details?id=fr.dvilleneuve.lockito

@0709wiwiwi seems to be using it in tests.

**Reply by bios4** [reply]
Thanks for the tip with Lockito 👍 - seems to work like a charm...

**Comment by DIrgi** [comment]
I have to say you did a really great job so far by establishing the wiki pages.
I just had a short visit. There are already lot of points written down and available I didn't expect..

Great job - Thanky to Emux, MotoUK and to the others who already added some parts.
![image](https://github.com/devemux86/cruiser/assets/57968526/83d63fee-74f6-4924-8d00-4c561f9aecea)

**Comment by bios4** [comment]
I added the Wiki page for the "[My collection menu](https://github.com/devemux86/cruiser/wiki/My-collection-menu)" today.

The descriptions are as current as of [Cruiser 5.3.0-b5](https://github.com/devemux86/cruiser/discussions/734#discussioncomment-12510204) from today incl. the swipe-deletion update.

**Reply by MotoUKRider** [reply]
@bios4 That's nice work, very extensive :heavy_check_mark:

**Reply by devemux86** [reply]
@bios4 Thanks, that's impressive work! :+1:

**Comment by bios4** [comment]
Just added the "Search settings" Wiki page.

One thing is totally left open: what does the "Offline search (zoom)" setting exactly do @devemux86?

And why are "GraphHopper" and "OpenRouteService" listed in the search settings?
Aren't they responsible for routing (like BRouter)? What are the benefits of either service?

Please let me know so that I can add this info, too...

**Reply by devemux86** [reply]
OpenRouteService and GraphHopper are also providers for online search.

The "Offline search (zoom)" enables the offline search from the selected zoom.
Offline search on the phone is not fast if we try to search whole country areas.
It's better to offline search when zooming on specific regions.

**Reply by bios4** [reply]
Thanks for the explanation!

**Reply by devemux86** [reply]
@bios4 Offline search is done inside the visible map area.

**Reply by MotoUKRider** [reply]
So, amusingly, you have to know where something is before you can search for it :laughing: 

It's not so bad if you that _Joe's Bar n Grill_ is somewhere in Manchester, but just make sure that the map is in that location.

**Reply by bios4** [reply]
> @bios4 Offline search is done inside the visible map area.

Huh?
Why do we need an additional zoom setting then?

**Reply by devemux86** [reply]
> Why do we need an additional zoom setting then?

@bios4 Because the default is not enough for some users.

**Reply by devemux86** [reply]
@MotoUKRider Offline search is done on the phone using the mobile cpu.

Germany, France, UK and other large countries have tens of millions of POIs.
Searching all of them for e.g. "cafe" takes too much time and users will complain.

That's why the zoom setting exists. If users want to search their whole country,
set the zoom to a small value and use the offline search from this small zoom.

But most offline searches are usually done over a zoomed city region.
So the default zoom 12 should be enough.

**Reply by bios4** [reply]
To sum it up I understand that it means that offline search is done

- **only** in the currently visible map area **_AND_**
- **only** when zoomed to the level defined in the settings (or zoomed-in higher)

Is that what you're saying?

**Reply by devemux86** [reply]
And some users have activated multiple maps, Germany + France + ...

i.e. search half of Europe for a cafe on the phone

**Reply by devemux86** [reply]
> To sum it up I understand that it means that offline search is done

@bios4 Correctly.

**Reply by bios4** [reply]
Finished the Wiki page for [Search settings](https://github.com/devemux86/cruiser/wiki/Search-settings) and updated the page for [Location settings](https://github.com/devemux86/cruiser/wiki/Location-settings) (initial work was done by @r11d11) in the same style...

**Reply by r11d11** [reply]
@bios4 Nice work 👍

**Reply by devemux86** [reply]
@bios4 Thanks, very nice work! :+1:

**Comment by MotoUKRider** [comment]
I have added a few words and 2 sample images to [the Navigation Guide page](https://github.com/devemux86/cruiser/wiki/Navigation-guide). I just wanted to get the images up as there was previously a blank page.

**Comment by bios4** [comment]
A few more additions:

- finished the [Map settings](https://github.com/devemux86/cruiser/wiki/Map-settings) page
- added the [Subscriptions](https://github.com/devemux86/cruiser/wiki/Subscriptions) page
- started working on the [Favorites guide](https://github.com/devemux86/cruiser/wiki/Favorites-guide)

The latter is "work-in-progress," but I won't have time to continue working on it until beginning/mid of next week...

**Reply by devemux86** [reply]
@bios4 Thank you for your continued impressive work on the Wiki! :+1:

**Reply by bios4** [reply]
You're welcome - I'm glad to be able to make at least a small contribution... 😃
   --> Extended the [Favorites guide](https://github.com/devemux86/cruiser/wiki/Favorites-guide) a little again.

**Comment by bios4** [comment]
The [GPS recording settings](https://github.com/devemux86/cruiser/wiki/GPS-recording-settings) page is also available as of today.

**Reply by devemux86** [reply]
@bios4 Many thanks! :+1:

**Comment by bios4** [comment]
Made again a little progress and started upating the [Routing settings](https://github.com/devemux86/cruiser/wiki/Routing-settings) page.
The initial work done by @MotoUKRider was merged...

I already prepared (all?) the required screenshots for that chapter. So if anyone wants to join in and continue/expand the documentation - go ahead! 😉

**Reply by devemux86** [reply]
@bios4 Thanks, very nice work! :+1:

**Comment by MotoUKRider** [comment]
I added a bit of text in the middle of the [Routing settings](https://github.com/devemux86/cruiser/wiki/Routing-settings) page about "renaming of routing profiles" and how they can help with different types of journey/ride.

For example, here's how I differentiate between "Fastest" and "No Motorways/Tolls" - a different route colour will help show routing profile I am using (either for the whole route or that particular segment)...

![cruiser-routing-examples-clip](https://github.com/user-attachments/assets/d4e2fb6b-a62a-47e1-86c7-08e9f2afaa9d)

(the full image is on the Wiki page linked above)

**Reply by devemux86** [reply]
@MotoUKRider Thanks! :+1:

**Comment by bios4** [comment]
I had a little time and updated all the existing Wiki pages for "[settings](https://github.com/devemux86/cruiser/wiki/Settings-info)" with current screenshots from Cruiser 5.7.4 and explained a few of the new options. The "[Voice guidance settings](https://github.com/devemux86/cruiser/wiki/Voice-guidance-settings)" page was filled with a bit of content.

The "[My collection menu](https://github.com/devemux86/cruiser/wiki/My-collection-menu)" and "[Favorites guide](https://github.com/devemux86/cruiser/wiki/Favorites-guide)" pages haven't been updated with new screenshots yet due to limited time, but a note was added already some time ago stating that the "tile" menus are no longer available in the current version of Cruiser.

It's a shame that there's almost no progress in building the Wiki.\
And yes, this criticism is also directed at myself...

**Reply by bios4** [reply]
@devemux86 - could you please think of some clarifying wording for the two "Distance from route" options in [ "Settings" ] > [ "Search" ] > [ "[Advanced](https://github.com/devemux86/cruiser/wiki/Search-settings#advanced)" ]?

What is the first option related to - POI?\
The second - favorites?

IMHO it would make sense to add its purposes to the description and not let users guess only by the icon...

And please let me know briefly what the option [ “Settings” ] > [ “Location” ] > [ “[Location orientation](https://github.com/devemux86/cruiser/wiki/Location-settings#location-orientation)” ] is used for so that I can add this info to the Wiki, too.

Thanks!

**Reply by devemux86** [reply]
@bios4 Thanks! :+1: 

> Settings | Search | Advanced | Distance from route

It is the distance from the route along which POI or favorites are searched, e.g. as displayed in navigation panels.

> Settings | Location | Location orientation

The location arrow gets its orientation by the movement direction or compass direction.

There are similar "orientation" settings:
- "Settings | Map | Map orientation": orientation of map, e.g. in follow mode

- "Settings | Navigation | Navigation orientation": orientation of navigation

**Reply by bios4** [reply]
> It is the distance from the route along which POI or favorites are searched, e.g. as displayed in navigation panels.

Sorry @devemux86, I guess I didn't express myself clearly in my post...

The meaning of these settings is clear to me and already described in the Wiki!

What I wanted is to ask you to update the descriptions of these separate settings in the app UI in any of the next versions so that its clear on first sight which setting represents what. Currently, both are named "Distance from route" with just a (not-so-self-explaining) different icon:

<img width="33%" src="https://github.com/user-attachments/assets/0047fc8e-628a-4380-9d13-5d57de73df5f" />

What is the first distance - POI?\
What is the second distance - favorites?

**Reply by devemux86** [reply]
@bios4 Your image is correct.

> update the descriptions of these separate settings in the app UI

We can do this.

**Reply by bios4** [reply]
The [Favorites guide](https://github.com/devemux86/cruiser/wiki/Favorites-guide) page was updated with new screenshots, too and the descriptions adapted to the "non-tiles" UI.

**Reply by bios4** [reply]
Finally I had some time left to update also the [My collection menu](https://github.com/devemux86/cruiser/wiki/My-collection-menu) page to the current "non-tiles" UI.
The screenshots and descriptions are all still based on Cruiser 5.7.4...
